tests/app/test_jinja_filters.py

```diff
@@ -95,6 +96,28 @@ def test_format_date(self):
 
         self.assertEqual(format_value, '1 January 2017')
 
+    def test_format_conditional_date(self):
+        # Given
+        date = '03/11/2017'
+        date2 = '04/08/2018'
+        dates = ([date, date2])
+
+        # When
+        format_value = format_conditional_date(dates)
+
+        self.assertEqual(format_value, '11 March 2017')
+
+    def test_format_conditional_date2(self):
```

Comment:
The method name should reflect what the method is testing, so something like `test_format_conditional_date_first_arg_none` would be better. 

It might also be worth considering creating a data structure to represent the test variations and loop over it e.g. 
```
test_dates = [
  ['03/11/2017', '04/08/2018', '11 March 2017'],
  [None, '10/12/17', '12 October 2017']
]
```
or possibly clearer using dicts:
```
test_dates = [
  {
    'date1': '03/11/2017', 
    'date2': '04/08/2018', 
    'result': '11 March 2017'
  },
  {
    'date1': None, 
    'date2': '10/12/17', 
    'result': '12 October 2017'
  }
]
```